### DISCUSSION TITLE
Functions concerned with Eclipse

### DISCUSSION THREAD

**Discussion by homedry** [ivoyager/forum#12] [Proposals]
### Describe what is missing or deficient and why it should be included or improved.

Simulating past and future eclipses is sometimes important for either academically or for fun.

### If this is a simulation or planetarium "content" proposal, describe how the content should appear and how users should interact with it.

1. User can see the Earth shaded by the Moon and the Moon by the earth if it is the time when an eclipse should occur
2. User can see an eclipse directly in a point of view of the Earth, like we sometimes see it on the Earth when the event happens

### Which of the following describes your feature? (1) A feature or content improvement for our Planetarium. (2) An improvement for game/app developers that should be in our core repositories. (3) An improvement for game/app developers that should be an optional addon.

Firstly and essentially (1) and possibly (2)

**Comment by charliewhitfield** [comment]
Sounds great!

We need to fix the [lack of shadows bug](https://github.com/ivoyager/ivoyager_core/issues/1). I keep revisiting that but I might need someone more familiar with Godot 3D rendering to look at it. If shadows were working then eclipses should happen and would be a great test of simulation accuracy.

It would also be helpful to have a view from Earth's surface outward, so you would have the perspective we have with sun and moon near the same size. (I've played with an "outward" view a long time ago. But my implementation wasn't so good so I dropped it.)
